src/webots/nodes/utils/WbObjectDetection.cpp

```diff
@@ -189,7 +189,7 @@ bool WbObjectDetection::isWithinBounds(const WbAffinePlane *frustumPlanes, const
         const WbIndexedFaceSet *indexedFaceSet = static_cast<const WbIndexedFaceSet *>(boundingObject);
         const WbCoordinate *coordinates = indexedFaceSet->coord();
         for (int i = 0; i < coordinates->pointSize(); ++i) {
-          points.append(objectRotation * (coordinates->point(i) * mObject->absoluteScale()) + objectPosition);
+          points.append(objectRotation * coordinates->point(i) + objectPosition);
```

Comment:
It would need to be tested to be 100% sure but by taking a quick look,  `mObject` is of `WbSolid` type (already was in master) therefore in theory there cannot be a Transform node in its ancestry so it should not be possible for it to have a scale. 
In practice however it's possible this was changed prior to the re-implementation a `absoluteScale` at the Pose level which turned out to be necessary only at a later stage of the rework, so technically it would still compile if the change is reverted.

The question therefore is whether it's necessary or not, I don't recall exactly all the cases where the objectdetection class is involved (like, does it need to have recognitioncolors or radarCrossSection, things like that?) If so those are only available to solids indeed, pure geometries wouldn't

Reply:
The mObject is always of type Solid.
But descandant geometries of mObject could still have parent Transform node.

However, I double checked and we are looking only in boundingObject field, so the transform case and scale can be safely ignored.